docs/concepts.md

```diff
@@ -35,91 +46,102 @@ When there is late arriving data (data intended for 9:00 arriving >1 hr late at
+ - **Read Optimized View** : Queries on this view see the latest snapshot of the dataset as of a given commit or compaction action. 
+    The view exposes only the base/columnar files in latest file slices to the queries and guarantees the same columnar query performance compared to a non-hudi dataset. 
+ - **Incremental View** : Queries on this view only see new data written to the dataset, since a given commit/compaction. This view effectively provides change streams to enable incremental data pipelines. 
+ - **Realtime View** : Queries on this view see the latest snapshot of dataset as of a given delta commit action. The view provides near-real time datasets in few mins, 
+    by merging log and base files of the latest file slice on-the-fly.
 
-Each record is uniquely identified by a `record key` and mapped to a file id forever. This mapping between record key
-and file id, never changes once the first version of a record has been written to a file. In short, the
- `file id` identifies a group of files, that contain all versions of a group of records.
+Following table summarizes the trade-offs between the different views.
 
+| Trade-off | ReadOptimized | RealTime |
+|-------------- |------------------| ------------------|
+| Data Latency | Higher   | Lower |
+| Query Latency | Lower (raw columnar performance) | Higher (merge columnar + row based delta) |
 
-## Copy On Write
 
-As mentioned above, each commit on Copy On Write storage, produces new versions of files. In other words, we implicitly compact every
-commit, such that only columnar data exists. As a result, the write amplification (number of bytes written for 1 byte of incoming data)
- is much higher, where read amplification is close to zero. This is a much desired property for a system like Hadoop, which is predominantly read-heavy.
+## Copy On Write Storage
+
+File slices in Copy-On-Write storage only contain the base/columnar file and each commit produces new versions of base files. 
+In other words, we implicitly compact on every commit, such that only columnar data exists. As a result, the write amplification 
+(number of bytes written for 1 byte of incoming data) is much higher, where read amplification is zero. 
+This is a much desired property for analytical workloads, which is predominantly read-heavy.
 
 Following illustrates how this works conceptually, when  data written into copy-on-write storage  and two queries running on top of it.
 
 
 {% include image.html file="hudi_cow.png" alt="hudi_cow.png" %}
 
 
-As data gets written, updates to existing file ids, produce a new version for that file id stamped with the commit and
-inserts allocate a new file id and write its first version for that file id. These file versions and their commits are color coded above.
+As data gets written, updates to existing file groups, produce a new slice for that file group stamped with the commit and
```

Comment:
As data gets written, updates to an existing file group produces a new slice for that file group stamped with the commit instant while inserts allocate a new file group and write its first version for that file group. These file slices and their commit instant times are color coded above.